packages/components/src/disabled/test/index.tsx

```diff
@@ -0,0 +1,153 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import Disabled from '../';
+
+jest.mock( '@wordpress/dom', () => {
+	const focus = jest.requireActual( '../../../../dom/src' ).focus;
+	return {
+		focus: {
+			...focus,
+			focusable: {
+				...focus.focusable,
+				find( context: Element, options = { sequential: false } ) {
+					// In JSDOM, all elements have zero'd widths and height.
+					// This is a metric for focusable's `isVisible`, so find
+					// and apply an arbitrary non-zero width.
+					Array.from( context.querySelectorAll( '*' ) ).forEach(
+						( element ) => {
+							Object.defineProperties( element, {
+								offsetWidth: {
+									get: () => 1,
+								},
+							} );
+						}
+					);
+
+					return focus.focusable.find( context, options );
+				},
+			},
+		},
+	};
+} );
+
+describe( 'Disabled', () => {
+	const Form = () => (
+		<form>
+			<input />
+			<div contentEditable tabIndex={ 0 } />
+		</form>
+	);
+
+	it( 'will disable all fields', () => {
+		const { container } = render(
+			<Disabled>
+				<Form />
+			</Disabled>
+		);
+
+		const input = container.querySelector( 'form input' );
+		const div = container.querySelector( 'form div' );
+		expect( input ).toBeDisabled();
+		expect( div ).toHaveAttribute( 'contenteditable', 'false' );
+		expect( div ).not.toHaveAttribute( 'tabindex' );
+		expect( div ).not.toHaveAttribute( 'disabled' );
+	} );
+
+	it( 'should cleanly un-disable via reconciliation', () => {
+		const MaybeDisable = ( { isDisabled = true } ) =>
+			isDisabled ? (
+				<Disabled>
+					<Form />
+				</Disabled>
+			) : (
+				<Form />
+			);
+
+		const { container, rerender } = render( <MaybeDisable /> );
+		rerender( <MaybeDisable isDisabled={ false } /> );
+
+		const input = container.querySelector( 'form input' );
+		const div = container.querySelector( 'form div' );
```

Comment:
Let's add a few checks here to show that we expect the form elements to be disabled:

```suggestion
		const input = screen.getByRole( 'textbox' );
		const contentEditable = screen.getByTitle( 'edit my content' );
		
		expect( input ).toBeDisabled();
		expect( contentEditable ).toHaveAttribute( 'contenteditable', 'false' );

		rerender( <MaybeDisable isDisabled={ false } /> );
```